File: Leet Code Solutions/2009. Minimum Number of Operations to Make Array Continuous/PralinKhaira-ContinousArray.js
var minOperations = function(nums) {
    nums.sort(function(a, b) { return a - b; });
    let n = nums.length;
    nums = [...new Set(nums)]; 

    let ans = Number.MAX_SAFE_INTEGER;

    for (let i = 0; i < nums.length; i++) {
        let s = nums[i];
        let e = s + n - 1;
        let idx = binarySearch(nums, e);

        ans = Math.min(ans, n - (idx - i));
    }

    return ans;
};

function binarySearch(nums, target) {
    let left = 0;
    let right = nums.length - 1;

    while (left <= right) {
        let mid = left + Math.floor((right - left) / 2);

        if (nums[mid] <= target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    return left;
}

File: Leet Code Solutions/2009. Minimum Number of Operations to Make Array Continuous/PralinKhaira-ContinousArray.cs
public class Solution {
    public int MinOperations(int[] nums) {
        Array.Sort(nums);
        List<int> list = new List<int>();
        
        for (int i = 0; i < nums.Length; i++)
        {
            if (i > 0 && nums[i] == nums[i - 1])
                continue;
            
            list.Add(nums[i]);
        }

        int max = 0;
        
        for (int i = 0; i < list.Count; i++)
        {
            int target = list[i] + nums.Length - 1;
            int index = list.BinarySearch(target);
            
            if (index < 0)
                index = (~index) - 1;
            
            max = Math.Max(max, index - i + 1);
        }
        
        return nums.Length - max;
    }
}